Source/p_map.c

```diff
@@ -1021,12 +1063,31 @@ static void P_HitSlideLine(line_t *ld)
   // Check for the special cases of horz or vert walls.
 
   // killough 10/98: only bounce if hit hard (prevents wobbling)
-  icyfloor = 
-    (demo_version >= 203 ? 
-     P_AproxDistance(tmxmove, tmymove) > 4*FRACUNIT : !compatibility) &&
-    variable_friction &&  // killough 8/28/98: calc friction on demand
-    slidemo->z <= slidemo->floorz &&
-    P_GetFriction(slidemo, NULL) > ORIG_FRICTION;
+  // icyfloor = 
+  //   (demo_version >= 203 ? 
+  //    P_AproxDistance(tmxmove, tmymove) > 4*FRACUNIT : !compatibility) &&
+  //   variable_friction &&  // killough 8/28/98: calc friction on demand
+  //   slidemo->z <= slidemo->floorz &&
+  //   P_GetFriction(slidemo, NULL) > ORIG_FRICTION;
+
+  //e6y
+  if (demo_version >= 203)
+  {
+    icyfloor =
```

Comment:
Most of this part of the code is actually identical to the original. If you keep indentation, there's probably only two conditions to change.